flaky TestHTTPRequestAfterGracefulShutdown

```
=== RUN   TestHTTPRequestAfterGracefulShutdown/Request.GetBody_not_set
    http_test.go:1199: 
        	Error Trace:	/home/runner/work/quic-go/quic-go/integrationtests/self/http_test.go:1199
        	            				/home/runner/work/quic-go/quic-go/integrationtests/self/http_test.go:1108
        	Error:      	An error is expected but got nil.
        	Test:       	TestHTTPRequestAfterGracefulShutdown/Request.GetBody_not_set
--- FAIL: TestHTTPRequestAfterGracefulShutdown (0.01s)
    --- PASS: TestHTTPRequestAfterGracefulShutdown/Request.GetBody_set (0.00s)
    --- FAIL: TestHTTPRequestAfterGracefulShutdown/Request.GetBody_not_set (0.00s)

--- a/integrationtests/self/http_test.go
+++ b/integrationtests/self/http_test.go
@@ -1164,7 +1164,7 @@ func testHTTPRequestAfterGracefulShutdown(t *testing.T, setGetBody bool) {
 	cl := &http.Client{Transport: tr}
 
 	// first request to establish the connection
-	resp, err := cl.Get(fmt.Sprintf("https://localhost:%d/", ln.Addr().(*net.UDPAddr).Port))
+	resp, err := cl.Get(fmt.Sprintf("https://%s/", proxy.LocalAddr()))
 	require.NoError(t, err)
 	require.Equal(t, http.StatusNotFound, resp.StatusCode)
 
@@ -1176,7 +1176,7 @@ func testHTTPRequestAfterGracefulShutdown(t *testing.T, setGetBody bool) {
 			WroteHeaders: func() { headersCount++ },
 		}),
 		http.MethodGet,
-		fmt.Sprintf("https://localhost:%d/echo", ln.Addr().(*net.UDPAddr).Port),
+		fmt.Sprintf("https://%s/echo", proxy.LocalAddr()),
 		io.LimitReader(strings.NewReader("foobar"), 1000),
 	)
 	require.NoError(t, err)

